src/lib/ChecksummedHex.ts

```diff
@@ -0,0 +1,98 @@
+//! Checksummed hex encoding following an [EIP-55][1]-like scheme.
+//!
+//! [1]: https://eips.ethereum.org/EIPS/eip-55
+
+//! Migrate from https://github.com/casper-network/casper-node/blob/9609a616439de334d89917a14240940615a49f81/types/src/checksummed_hex.rs
+
+import { concat } from '@ethersproject/bytes';
+import { byteHash } from './ByteConverters';
+import { decodeBase16 } from './Conversions';
+
+// The number of input bytes, at or below which [`decode`] will checksum-decode the output.
+export const SMALL_BYTES_COUNT = 75;
+
+/**
+ * Takes a slice of bytes and breaks it up into a vector of *nibbles* (ie, 4-bit values)
+ * @param bytes
+ * @returns double sized of Unit8Array
+ */
+const bytesToNibbles = (bytes: Uint8Array): Uint8Array => {
+  let outputNibbles = new Uint8Array();
+  bytes.forEach(byte => {
+    outputNibbles = concat([
+      outputNibbles,
+      Uint8Array.of(byte >>> 4, byte & 0x0f)
+    ]);
+  });
```

Comment:
do you think you can switch from `forEach` to `reduce` with new empty Uint8Array here?